### PR TITLE
Constant->Result networks

### PR DESCRIPTION
**Pull request by ilyachur**

### CODE REVIEW

**Comment by ilyachur** [comment]
@AlexPeskov In this PR I am changing the iterator to support constant networks.
Also I add new functional tests but disabled it because at the current moment CPU plugin doesn't support such behaviour.

File: inference-engine/src/legacy_api/include/legacy/details/ie_cnn_network_iterator.hpp [hunk 1]
```diff
@@ -32,25 +33,87 @@ CNNNetworkIterator {
...
+        auto get_consumers = [](const CNNLayerPtr& node) -> std::vector<CNNLayerPtr> {
+            std::vector<CNNLayerPtr> consumers;
+            for (const auto & output : node->outData) {
+                for (const auto &consumer : getInputTo(output)) {
+                    consumers.push_back(consumer.second);
+                }
             }
+            return consumers;
+        };
+        auto bfs = [&](const CNNLayerPtr& start_node, bool traverse_via_outputs = false) {
```
Comment (esmirno) [inline inference-engine/src/legacy_api/include/legacy/details/ie_cnn_network_iterator.hpp:60]: if this bfs ported from v7 serializer ? as i remember it never worked in real networks, always missed some layers in iterations.

Comment (ilyachur) [reply inference-engine/src/legacy_api/include/legacy/details/ie_cnn_network_iterator.hpp:60]: @esmirno I took it from the `SortTopologically` function. As I know we actively use it in the IE plugins.

Comment (ilyachur) [reply inference-engine/src/legacy_api/include/legacy/details/ie_cnn_network_iterator.hpp:59]: Moreover, as I remember @GlebKazantaev fixed this method.

**Comment by ilyachur** [comment]
@esmirno @GlebKazantaev @ilya-lavrenov Please take a look.

**Review by GlebKazantaev — APPROVED** [review]

**Pull request merged**